### PR TITLE
fix catalog generation

### PR DESCRIPTION
**Pull request by drewbanin**
Fixes some issues around catalog generation/operations

**Changes**:
- Do not add `this` to the context for Operations
- Only add operations to the graph once. Previously, they were added once a Macro node, and once as an Operation node. Similarly, Operation nodes were previously created for code declared in `{% macro %}` blocks.
- Fix missing parent_map and child_map entries. This happened because the manifest was written out before refs were processed
- Run the compilation step before writing out the catalog for `dbt docs generate`. This simplifies the code a whole lot

### CODE REVIEW

File: dbt/context/common.py [hunk 1]
```diff
@@ -341,7 +340,32 @@ class AdapterWithContext(adapter_type):
...
+    pre_hooks = node.get('config', {}).get('pre-hook')
+    post_hooks = node.get('config', {}).get('post-hook')
+
+    model_context = {
+        "post_hooks": post_hooks,
+        "pre_hooks": pre_hooks,
+        "model": node,
+        "sql": node.get('injected_sql'),
+        "this": get_this_relation(db_wrapper, project_cfg, profile, node),
+        "ref": provider.ref(db_wrapper, node, project_cfg, profile, flat_graph)
```
Comment (drewbanin) [inline dbt/context/common.py]: `ref`, `sql` should be in the global context

File: dbt/loader.py [hunk 2]
```diff
@@ -17,7 +17,9 @@ def load_all(cls, root_project, all_projects):
         for loader in cls._LOADERS:
             nodes.update(loader.load_all(root_project, all_projects, macros))
 
-        return ParsedManifest(nodes=nodes, macros=macros)
+        manifest = ParsedManifest(nodes=nodes, macros=macros)
```
Comment (drewbanin) [inline dbt/loader.py]: tell the ParsedManifest to process itself:
 - process refs
 - incorporate Schema Spec info


UnparsedManifest.parse() ---> ParsedManifest

**Comment by cmcarthur** [comment]
given the changes on this plus what i sent you about parsing, maybe we should hold off on this. it's not actually causing problems rn, right?

**Comment by cmcarthur** [comment]
@drewbanin ok, yeah that sounds fine

**Comment by drewbanin** [comment]
@cmcarthur I made some other changes around operations in this PR. Going to rip them out into a different PR, then close this one.

File: dbt/loader.py (hunk 2 above)
Comment (cmcarthur) [reply dbt/loader.py]: i've started an issue for this here: https://github.com/fishtown-analytics/dbt/issues/821

**Pull request merged**